test/ZoraNFTCreatorV1_Fork.t.sol

```diff
@@ -178,4 +189,45 @@ contract ZoraNFTCreatorV1Test is Test, ForkHelper {
         drop.purchase{value: fee}(1);
         assertEq(drop.tokenURI(1), "DEMO");
     }
+
+    function test_ForkUpgradeWithDifferentContractName() external {
+        string[] memory forkTestChains = getForkTestChains();
+
+        for (uint256 i = 0; i < forkTestChains.length; i++) {
+            string memory chainName = forkTestChains[i];
+            vm.createSelectFork(vm.rpcUrl(chainName));
+            creator = ZoraNFTCreatorV1(getDeployment().factory);
+            verifyAddressesFork(chainName);
+            deployCore();
+            forkContractNameWithRevert();
+            forkContractNameSucceed();
+        }
+    }
+
+    function forkContractNameWithRevert() internal {
+        ZoraNFTCreatorV1 newCreatorImpl = new ZoraNFTCreatorV1(address(dropImpl), editionMetadataRenderer, dropMetadataRenderer);
+        address owner = creator.owner();
+
+        vm.prank(owner);
+        creator.upgradeTo(address(newCreatorImpl));
+
+        MockContractMetadata mockMetadata = new MockContractMetadata("uri", "test name");
+
+        vm.prank(owner);
+        vm.expectRevert(abi.encodeWithSignature("UpgradeToMismatchedContractName(string,string)", "ZORA NFT Creator", "test name"));
+        creator.upgradeTo(address(mockMetadata));
+    }
+
+    function forkContractNameSucceed() internal {
```

Comment:
same here above regarding contract name, could the test name be more descriptive/clear, such as `forkUpgradeSucceedsWithGoodContractName`